### PR TITLE
Reduce overhead in ZoneAwareLoadBalancerBase by fixing alignment

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Reduce overhead in ZoneAwareLoadBalancerBase by fixing alignment
Additional Description: Moves many of the smaller members of the class to the end to save small amounts of memory
Risk Level: Low

### CODE REVIEW

File: source/common/upstream/load_balancer_impl.h
Comment (adisuissa) [inline source/common/upstream/load_balancer_impl.h:434]: nit: please add a small note about the coalescing of these data members at the end (something like [this](https://github.com/envoyproxy/envoy/blob/1ac0a5948ca317fd418777dab2ffe27935c0a276/source/common/router/config_impl.h#L342)).

**Review by adisuissa — COMMENTED** [review]
Thanks!
LGTM, modulo small nit.

File: source/common/upstream/load_balancer_impl.h
Comment (DiazAlan) [reply source/common/upstream/load_balancer_impl.h:435]: Done

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

/assign-from @envoyproxy/senior-maintainers

**Review by alyssawilk — APPROVED** [review]
Nice change!

**Pull request merged**
